Runtime/Connection/SerialDevice/SerialDeviceConnection.cs

```diff
@@ -147,82 +148,101 @@ public override void StopReading()
             StopReadPollingAsync();
         }
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        Stream EnsureTouchPanelSerialStreamIsOpen(SerialPort serialSession)
+        void EnsureSerialPortIsOpen(SerialPort serialSession)
         {
-            if (serialSession.IsOpen)
+            if (!serialSession.IsOpen)
             {
-                return serialSession.BaseStream;
+                serialSession.Open();
+                _device.OnConnected();
+            }
+        }
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        void ReadFromSerialPort(SerialPort serialPort,ReceiveDataHandler readDataCallback)
+        {
+            var bytes2Read = _serialPort.BytesToRead;
+            if (bytes2Read == 0)

```

Comment:
please maintain style of using braces {} with conditionals, braces should always be used with if statements to prevent preventable bugs.